Expanded data structure and formats page + a small things (#57)
nsidc/NSIDC-Data-Cookbook: reference-guides/data-structures.qmd

It is helpful to distinguish between:

**Data structure** — how data observations are organized conceptually (e.g. as a grid, rows/columns, discrete features, etc.), and  

**File format** — how those structured data are stored and delivered. 

Understanding the basics (and differences between the two) can help you:

- Interpret results correctly  
- Recognize limitations  
- Avoid common analysis errors, and 
- Do science more efficiently! 

In practice, the relationship between a science question, available data, and the tools used for analysis is often not linear. Science questions often guide data collection and product design, but they are also shaped by what data already exist. As a result, data structures, file formats, and tools are interconnected, with multiple valid pathways depending on the dataset and workflow. The figure below highlights these conceptual relationships between science questions, data structures, file formats, and analysis tools. 

```{mermaid}
flowchart TD

A[Science question]

B[Data structures]
B1[Raster]
B2[Swath]
B3[Vector]
B4[Tabular]

C[File formats]
C1[NetCDF / HDF5]
C2[GeoTIFF]
C3[Shapefile]
C4[CSV]

> D[Tools]

Reviewer: I think this figure is much improved with a better introduction! In fact, it's fine as is, but if one could move the "tools" a little further to the right or even up a bit to separate it more visually, that would help, but I don't know how easy that is to do.

D1[Python]
D2[R]
D3[GIS software]
D4[Web tools]

A -.-> B
B -.-> A

B --> B1
B --> B2
B --> B3
B --> B4

B1 --> C
B2 --> C
B3 --> C
B4 --> C

C --> C1
C --> C2
C --> C3
C --> C4

C -.-> D

D --> D1
D --> D2
D --> D3
D --> D4
```

## How Earth science data are structured

Before working with specific file formats, it helps to understand the **underlying data structures** they represent. At NSIDC, most datasets fall into one or more of the structures below.

### Raster / gridded data

::: {.callout-tip}
### At a glance... 
**Common applications:** Continuous spatial fields  
**Common formats:** GeoTIFF, NetCDF, HDF5, COG, ASCII grid
:::

For most remotely sensed imagery products, raster data represent space as a **regular grid of cells**. Each cell corresponds to a defined area on Earth and stores a value or category. For example, below you have a grid where each grid-cell (also called a pixel) contains a numeric value between 1 and 5.

```
+----+----+----+
|  3 |  4 |  5 |
+----+----+----+
|  2 |  3 |  4 |
+----+----+----+
|  1 |  2 |  3 |
+----+----+----+
```

> In this simplified example, each one of these cells represents a geographic footprint, not a point. Meaning, anywhere within one of these boxes (whether it is in the middle, or near the outer edge) is represented by the same numeric value. It is worth noting that this interpretaion can vary by product and methodology, where output from models is typically only valid for a point, and techniques for resampling remote sensing data can change the assumptions around grid-cell interpretation.

Reviewer 2: ```suggestion
In this simplified example, each one of these cells represents a geographic footprint, not a point. Meaning, anywhere within one of these boxes (whether it is in the middle, or near the outer edge) is represented by the same numeric value. It is worth noting that this interpretation can vary by product and methodology, where output from models is typically only valid for a point, and techniques for resampling remote sensing data can change the assumptions around grid-cell interpretation.
```

:::

#### NetCDF =  dimensions + variables + attributes

A NetCDF file typically contains three key components.

##### Dimensions define the **shape** of the data. Below is an example of a 3-Dimensional (3-D) data set.

```
Dimensions:
  time = 365
  y    = 720
  x    = 720
```

Each dimension represents an independent axis:

- `time` → when the observation occurred
- `y`, `x` → the coordinates of each grid cell (usually the center of the grid cell).
- sometimes `z` or `level` → vertical structure (this would be a 4th dimension)

##### Variables are **arrays of values** defined over one or more dimensions.

```
Variables:
  sea_ice_concentration(time, y, x)
  latitude(y, x)
  longitude(y, x)
```

Visually, you can think of a 3-dimensional variable (like `sea_ice_concentration` above) as a stack of 2-D maps through time that form a data-cube:

> ```

Reviewer 2: I like the schematics/visuals, but the ability of ascii characters to represent a 3-d concept feels limiting here. So could be replaced with something like this: https://www.svgviewer.dev/s/9ZojmQyT

(drawing direct inspiration the images here: https://desktop.arcgis.com/en/arcmap/latest/manage-data/netcdf/fundamentals-of-netcdf-data-storage.htm)

Reviewer: Agreed, but what is here works as a decent place-holder in the interim. :)

time = t0      time = t1      time = t2
+----+----+    +----+----+    +----+----+
|    |    |    |    |    |    |    |    |
+----+----+    +----+----+    +----+----+
|    |    |    |    |    |    |    |    |
+----+----+    +----+----+    +----+----+
```

Each slice shares:

- the same grid
- the same projection
- the same variable definition

##### Attributes store **metadata** that describe how to interpret the values and may be attached to:

- the file (global attributes)
- individual variables
- spatial coordinate variables

> Examples include:

Reviewer: I find myself wanting to change this to, " Attribute examples include:"

This is because the sections for Dimensions, Variables and Attributes aren't visually obvious. Perhaps some kind of styling could help here. The ##### level may help with screen readers, but visually, it isn't doing much. Maybe bold these three in these lines as well?

The above is a highly idealized example as swath geometry varies by mission, but notice how: 

- Spacing can vary across the swath (although this is not the case for all sensors)
- Footprints overlap differently near edges
- No fixed row/column alignment

Swath data typically preserves:

- Native resolution
- Viewing angle
- Exact observation timing

This makes them important for:

- Calibration and validation
- Algorithm development
- Applications sensitive to observation geometry

#### Common challenges
- Harder to visualize
- Not directly comparable across time or space
- Must be transformed to a regular geospatial grid before analysis

::: {.callout-note}
### More about the HDF-EOS data format...

HDF-EOS adds Earth Observing System conventions on top of HDF to represent satellite data structures such as swaths, grids, and point measurements. It was created so NASA mission products could store observation geometry and geolocation alongside measurements in a consistent way.
:::

### Tabular data 
::: {.callout-tip}
### At a glance... 

**Common applications:** Discrete observations, station data, or summaries  
**Common formats:** CSV, TSV, Parquet, NetCDF tables
:::

> For many users, tabular data feels familiar because it resembles spreadsheets used in tools like Excel or Google Sheets.

Reviewer: To orient the reader to the "more" box below, have this say, "For many users, tablular or delimited data feels..."

Otherwise, "delimited" first shows up in that box.

A spreadsheet may show *what* the values are, but not *where* or *how* they were produced.

In general, tabular structures are well-suited for:

- Single-location time series
- Small collections of points
- Summaries and diagnostics

But are less suitable for:

- Continuous spatial fields
- Large-scale gridded products
- Analyses requiring detailed spatial context

::: {.callout-note}
### More about delimited text formats...

Delimited text formats like CSV were created to provide a simple, universal way to store tabular data that can be opened in nearly any software environment. Because they are human-readable and easy to generate, they remain widely used for time series, summaries, and data exchange.
:::

### Vector spatial data 

::: {.callout-tip}
### At a glance... 

**Common applications:** Boundaries, tracks, and discrete geographic features  
**Common formats:** Shapefile, GeoJSON, GeoPackage, KML
:::

Vector data typically represent **discrete geographic features** rather than continuous spatial fields. 

```
• Point        ─── Line        ▢ Polygon
```

Examples:

- Glacier outlines
- Ice sheet boundaries
- Grounding lines

> #### Geometry types

Reviewer: Would be good to have images here eventually as well... especially for the polygons. Happy to create some!

##### Points
```
•
```

Points represent a single location with no area or length.

Common uses:

- Observation stations
- Sample locations
- Reference points

##### Lines
```
───────
```

Lines represent linear features with length but no area.

Common uses:

- Grounding lines
- Ice front positions
- Transects

##### Polygons
```
▢▢▢▢
▢▢▢▢
```

Polygons represent areas with defined boundaries.

Common uses:

- Glacier outlines
- Ice sheet masks
- Drainage basins

#### Attributes

Each feature includes attributed data, where attributes describe properties of **unique features**:

```
Feature ID | Name        | Area_km2 | Year
-----------|-------------|----------|------
001        | Glacier A   | 12.4     | 2020
002        | Glacier B   |  8.9     | 2020
```

Vector formats exist because many spatial features are:

- Discrete rather than continuous
- Better described by boundaries than averages
- Interpreted as objects, not fields

::: {.callout-note}
### More about Shapefile data format...

> The shapefile format was introduced as an early GIS standard for storing vector features with attributes. Although newer formats now exist, shapefiles remain common because of their compatibility across tools and legacy datasets.

Reviewer: You refer to this later, but I think it's good to mention the multi-file aspect of this format in this callout:

"The shapefile format, actually composed of multiple files, was introduced as..."

:::

#### How vector data differ from rasters

```
Raster:  value everywhere
Vector:  value only where features exist
```

Key differences:

- Geometry is explicit, not implied by grid cells
- Precision depends on digitization, not grid spacing
- Attributes are attached to features, not cells

Vector data represent **objects and boundaries**, not continuous fields.

## Common file formats at NSIDC

The table below lists common file formats encountered in NSIDC data products and some practical considerations when working with them.

| File format | Common uses | Quirks | NSIDC example products | Common tools (programmatic + GUI) | 
|------------|---------------|-------------|--------------------|------------------------|-----------------------------------|

> | **NetCDF4 / NetCDF classic** | Climate records, reanalysis products, gridded time series, model output. | Time encoding confusion; multidimensional indexing; large file sizes; steep learning curve without high-level tools. | [Sea Ice Concentration CDR](https://nsidc.org/data/G02202) • [Antarctic Ice Velocity](https://nsidc.org/data/NSIDC-0484)  | **Python:** `xarray`, `netCDF4` (low-level) • **R:** `terra`, `stars`, `ncdf4` • **GUI:** Panoply, QGIS, ArcGIS Pro, HDFView|

Reviewer 2: ```suggestion
| **NetCDF4 / NetCDF classic** | Climate records, reanalysis products, gridded time series, model output. | Time encoding confusion; multidimensional indexing; large file sizes; steep learning curve without high-level tools. | [Sea Ice Concentration CDR](https://nsidc.org/data/G02202) • [Antarctic Ice Velocity](https://nsidc.org/data/NSIDC-0484)  | **Python:** `xarray`, `netCDF4` (low-level)<br>**R:** `terra`, `stars`, `ncdf4`<br>**GUI:** Panoply, QGIS, ArcGIS Pro, HDFView|
```

Reviewer 2: Replacing the bullets with the line break `<br>` throughout the table (particularly for the "common tools" column) could make separation between the tools a little cleaner/easier to follow

| **HDF5** | Mission-level products, intermediate processing outputs, complex hierarchical datasets. | Deep hierarchies; unclear variable naming; inconsistent conventions between products. | [SMAP L3 Soil Moisture](https://nsidc.org/data/SPL3SMP) • [ICESat-2 ATL03](https://nsidc.org/data/ATL03) | **Python:** `rioxarray`, `h5py` • **R:** `rhdf5` • **GUI:** Panoply, HDFView|
| **HDF-EOS2 / HDF-EOS5** | Satellite swath and gridded mission products (e.g., MODIS). | Specialized structure; harder to read without EOS-aware tools; metadata spread across groups. | [MODIS Snow Cover](https://nsidc.org/data/mod10a1/) • [AMSR-E/AMSR2 SWE](https://nsidc.org/data/au_dysno/) | **Python:** `xarray`, `h5netcdf`, `pyhdf` • **R:** `terra` • **GUI:** Panoply, HDFView|
| **GeoTIFF / TIFF/ COG** | Gridded maps, spatial subsets, visualization-ready products. | Limited metadata compared to NetCDF; alignment issues across datasets; large files at high resolution. | [Ice Sheet Velocity](https://nsidc.org/data/nsidc-0731/) • [Land Freeze/Thaw Status](https://nsidc.org/data/nsidc-0728/) | **Python:** `rasterio`, `rioxarray`, `xarray` • **R:** `terra`, `raster` • **GUI:** QGIS, ArcGIS Pro |

> | **Shapefile** | Ice sheet boundaries, grounding lines, glacier outlines. | Multi-file dependency; field name limits; aging format. | [Glacier Database](https://nsidc.org/data/nsidc-0272/) • [Glacier Terminus Positions](https://nsidc.org/data/nsidc-0642/) | **Python:** `geopandas`, `shapely` • **R:** `sf` • **GUI:** QGIS, ArcGIS Pro |

Reviewer: This is nothing to resolve, just please please please someone tell me we have a non-shapefile vector dataset SOMEWHERE that we can point to so we can have more than shapefile in this list. I suppose we do have some flightline KMLs, so maybe those could be added, but I was hoping for geojsons.

---
title: "Data formats and structures"
---

Earth science and remote sensing data are stored in a variety of structures and file formats, each shaped by how the data were collected, processed, distributed, and historically used. At NSIDC, formats reflect satellite mission design, modeling needs, interoperability standards, and long-term archiving requirements.

> Research usually starts with a science question, not a format or structure preference. This page explains some common hurdles with specific data formats, formats that are most prevalent at NSIDC, and briefly describes the underlying data structures that some of these formats represent.

Reviewer 3: Suggest "This page describes the formats and structures used to store NSIDC data so that users can better understand how to work with this data."

I think the biggest "hurdle" is user's lack of understanding on the data formats.

Reviewer: I like Andy's suggestion for the "This page..." text. I would also start this paragraph with that and move the research question concept to the prior:

"Earth observation data are stored in a variety of structures
and file formats, each shaped by how the data were collected, processed,
distributed, and historically used. Research usually starts with a science question, not a format or structure preference. At NSIDC, formats reflect satellite mission
design, modeling needs, interoperability standards, and long-term archiving requirements."

It is helpful to distinguish between:

> **Data structure** — how observations relate to space, time, and each other  

Reviewer 3: I think **Data structure** relates to how the data are organized: as a table (e.g. rows and columns), as a 2D raster data, as multi-dimensional grids, or as vector data.  The coordinates (x, y, level, time) relate the data to space and time.

**File format** — how those structured data are stored and delivered 

> Formats and structure encode scientific assumptions and technical constraints within data products. Understanding the basics can help you:

Reviewer 3: It would help to add some examples of constraints and assumptions.  I am not entirely clear by what is meant here.

## Common file formats at NSIDC

| File format | Common uses | Quirks | NSIDC example products | Common tools (programmatic + GUI) | 
|------------|---------------|-------------|--------------------|------------------------|-----------------------------------|
| **NetCDF4 / NetCDF classic** | Climate records, reanalysis products, gridded time series, model output. | Time encoding confusion; multidimensional indexing; large file sizes; steep learning curve without high-level tools. | [Sea Ice Concentration CDR](https://nsidc.org/data/G02202) • [Antarctic Ice Velocity](https://nsidc.org/data/NSIDC-0484)  | **Python:** `xarray`, `dask`, `netCDF4` (low-level) • **R:** `terra`, `stars`, `ncdf4` • **GUI:** Panoply, QGIS|
| **HDF5** | Mission-level products, intermediate processing outputs, complex hierarchical datasets. | Deep hierarchies; unclear variable naming; inconsistent conventions between products. | [SMAP L3 Soil Moisture](https://nsidc.org/data/SPL3SMP) • [ICESat-2 ATL03](https://nsidc.org/data/ATL03) | **Python:** `xarray`, `pandas`, `h5py` (inspection) • **R:** `rhdf5` • **GUI:** HDFView|

> | **HDF-EOS2 / HDF-EOS5** | Satellite swath and gridded mission products (e.g., MODIS). | Specialized structure; harder to read without EOS-aware tools; metadata spread across groups. | [MODIS Snow Cover](https://nsidc.org/data/mod10a1/) • [AMSR-E/AMSR2 SWE](https://nsidc.org/data/au_dysno/) | **Python:** `xarray`, `h5netcdf`, `pyhdf` • **R:** `terra` • **GUI:** Panoply, HDFView|

Reviewer 3: I think you need `rioxarray` to read HDF-EOSx into `xarray` because the HDF-EOSx GDAL driver is used.

| **GeoTIFF / TIFF** | Gridded maps, spatial subsets, visualization-ready products. | Limited metadata compared to NetCDF; alignment issues across datasets; large files at high resolution. | [Ice Sheet Velocity](https://nsidc.org/data/nsidc-0731/) • [Land Freeze/Thaw Status](https://nsidc.org/data/nsidc-0728/) | **Python:** `rasterio`, `rioxarray`, `xarray` • **R:** `terra`, `raster` • **GUI:** QGIS, ArcGIS Pro |
| **Shapefile** | Ice sheet boundaries, grounding lines, glacier outlines. | Multi-file dependency; field name limits; aging format. | [Glacier Database](https://nsidc.org/data/nsidc-0272/) • [Glacier Terminus Positions](https://nsidc.org/data/nsidc-0642/) | **Python:** `geopandas`, `shapely` • **R:** `sf` • **GUI:** QGIS, ArcGIS Pro |

> | **CSV / TSV** | Time series, summaries, point extractions. | No spatial context; metadata easily lost; scaling issues. | [Ice Thickness](https://nsidc.org/data/irmcr2/) | **Python:** `pandas` • **R:** `readr`, `data.table`, `tibble` • **GUI:** Excel, LibreOffice |

Reviewer 3: There is spatial context if map coordinates are included, e.g. latitude and longitude columns.

> | **ASCII grid / text** | Historical products, simple model outputs. | No embedded CRS; manual parsing required. | [Sea Ice Trends](https://nsidc.org/data/nsidc-0192/) | **Python:** `numpy`, `pandas` • **R:** base R, `terra` • **GUI:** Text editors, QGIS (limited) |

Reviewer 4: `pandas` handles 2D arrays - is that different than a 2D grid?  I might be confused by terminology here.

| **PNG / GIF / JPEG** | Quicklook imagery, browse products, outreach. | Not analysis-ready; no quantitative metadata. | [Arctic Change](https://nsidc.org/data/nsidc-0589/) | **GUI:** Web browser, image viewers, NASA Worldview |
| **JSON** | Metadata responses, data access services. | Not suitable for large arrays; not analysis-ready. | [SMAP Ancillary Files](https://nsidc.org/data/smap_l4_sm_anc_log/) | **Python:** `json`, `requests` • **R:** `jsonlite` • **GUI:** Web browser |

## How Earth science data are structured

> Before working with specific file formats, it helps to understand the **underlying data structures** they represent. At NSIDC, most datasets fall into one or more of the structures below.

Reviewer 4: as mentioned above, should we put the table later then, since it helps to understand structure first?

## Raster / gridded data

::: {.callout-tip}
### At a glance... 
**Best for:** Continuous spatial fields  
**Common formats:** GeoTIFF, NetCDF, HDF5, COG, ASCII grid
:::

> Raster data represent space as a **regular grid of cells**. Each cell corresponds to a defined area on Earth and stores a value or category. For example, below you have a grid where each box represents a numeric value between 1 and 5. 

Reviewer 3: ```suggestion
Raster data represent space as a **regular grid of cells**. Each cell corresponds to a defined area on Earth and stores a value or category. For example, below you have a grid where each grid-cell (also called a pixel) contains a numeric value.
```

> ```

Reviewer 3: We should replace this with a figure.

```{python}
from itertools import product

import matplotlib.pyplot as plt
import numpy as np

fig, ax = plt.subplots(figsize=(5,5))

ax.set_ylim(0,3)
ax.set_xlim(0,3)
[ax.axhline(y, c='k') for y in np.arange(1,3)]
[ax.axvline(y, c='k') for y in np.arange(1,3)]

[
    ax.text(x, y, value, ha='center', va='center', fontsize=50) 
    for (x, y), value in zip(
        product(np.arange(0.5,3), np.arange(0.5,3)), 
        [3,4,5,2,3,4,1,2,3]
    )
]

ax.set_xticks([])
ax.set_yticks([]);
```

Reviewer: ![basic-raster-graphic](images/basic-raster-graphic.png)

I ran Andy's code and made this .png and can load this into the images folder if we want...

+----+----+----+
|  3 |  4 |  5 |
+----+----+----+
|  2 |  3 |  4 |
+----+----+----+
|  1 |  2 |  3 |
+----+----+----+
```

Each one of these cells represents a geographic footprint, not a point. Meaning, anywhere within one of these boxes (whether it is in the middle, or near the outer edge) is represented by the same numeric value.

### Key concepts
- **Extent** — the geographic area covered  

> - **Resolution** — the size of each cell (e.g., 25 km × 25 km)  

Reviewer 3: **Spatial Resolution** - of an image is not the same as pixel size.  Spatial resolution is the _spatial resolving power_ of the remote sensing system, which is related to the geometry and optics of the system, and also contrast in the image.  **Cell Size** is a better term.

- **Coordinate Reference System (CRS)** — how the grid maps to Earth  
- **Grid definition** — origin, orientation, and spacing  

::: {.callout-note}
### Why GeoTIFF exists

GeoTIFF extends standard image files by embedding geographic referencing information directly in the file.  This makes them portable across GIS tools without requiring separate projection or coordinate files. 
:::

### What raster values mean
Raster values usually represent either:

- a value recorded at / near the **cell center**, or  
- an **average or modeled value across the cell area**

This distinction matters for interpretation and comparison. These decisions affect uncertainty and are typically documented in product metadata rather than the file format itself.

> ### Where rasters come from

Reviewer 3: Suggest **Types of data represented by rasters"

This information could be included in the introduction rather than as a separate section.

Many raster products are **derived**, not directly observed. Examples of this would be:

- Satellite swaths being aggregated to a grid
- Multiple observations being averaged into one cell
- Point data interpolated or modeled onto a grid

::: {.callout-note}
### Why ASCII grids exist

ASCII grid formats were developed as simple, human-readable ways to store gridded data that could be shared across software systems. They remain common in legacy datasets because they are transparent and easy to parse, even though they lack embedded spatial metadata.
:::
 

> ### NetCDF data - a sometimes tricky container for raster data

Reviewer 3: I'm not sure we should have judgement statements here.  I think NetCDF, as with any other unfamiliar format, requires some time to understand the format and what tools to use to work with that format.  If users had to work with flat binary files they would see NetCDF and other self describing geospatial array formats as very easy.

Many NSIDC datasets are best understood as named, labeled arrays, not as tables or spreadsheets.

> Instead of rows and columns, NetCDF organizes data by dimensions such as space, time, and height.

Reviewer 3: This is not correct.  A variable stored in a NetCDF in raster formats - if more than 2 dimensions multiple rasters - is still stored as rows and columns.  What the NetCDF format (and also an HDF) allows is for coordinates to be assigned to those dimensions.  And also for those dimensions to be named.  It also allows attributes to be attached to each data variable and coordinate variable.  

I have a draft tutorial in progress that describes this.

###### 3-D (for example, a variable over time and horizontal space)
```
variable(time, y, x)
```

Examples:

- Sea ice concentration
- Snow cover
- Surface temperature

###### 4-D (perhaps a variable over time + horizontal space + vertical space)
```
variable(time, level, y, x)
```

Examples:

- Atmospheric pressure levels (e.g., 1000 hPa → 100 hPa)
- Model layers (terrain-following or topography)
- Depth or height above/ below a surface

## Swath data 

::: {.callout-tip}
### At a glance... 

**Best for:** Preserving original satellite observation geometry  
**Common formats:** HDF-EOS, NetCDF with geolocation arrays
:::

Swath data preserve measurements **close to their native satellite measurement geometry** along an orbit.

```
Satellite flight direction →
────────────────────────────────────

   ◯     ◯      ◯      ◯      ◯
     ◯      ◯      ◯      ◯
   ◯     ◯      ◯      ◯      ◯
     ◯      ◯      ◯      ◯
```
Each ◯ = one observation footprint

The above is a highly idealized example as swath geometry varies by mission, but notice how: 

- Spacing varies across the swath
- Footprints overlap differently near edges
- No fixed row/column alignment

### Why swaths exist

> Swath formats preserve:

Reviewer 3: I don't know of a swath format.  It is more of a type of data or a way data are structured.

Google's AI suggests this as a description:
>Satellite swath data represents the strip of Earth's surface imaged or scanned by a satellite sensor during a single orbital pass, typically tens to hundreds of kilometers wide. These data are often organized as non-rectilinear, time-ordered grids (swath coordinates) rather than map-projected grids, providing high-resolution,,,, native-resolution measurements for specific locations.

Which I think comes from this [page](https://natural-resources.canada.ca/maps-tools-publications/satellite-elevation-air-photos/satellite-characteristics-orbits-swaths)

This presentation from the HDF has some example of how swath data could be encoded: https://cfconventions.org/Meetings/2017-workshop/2_4_CF_Swath.pdf

- Native resolution
- Viewing angle
- Exact observation timing

This makes them important for:

- Calibration and validation
- Algorithm development
- Applications sensitive to observation geometry

::: {.callout-note}
### Why HDF-EOS exists

HDF-EOS adds Earth Observing System conventions on top of HDF to represent satellite data structures such as swaths, grids, and point measurements. It was created so NASA mission products could store observation geometry and geolocation alongside measurements in a consistent way.
:::

### Common challenges
- Harder to visualize
- Not directly comparable across time or space
- Usually must be transformed to a grid before analysis

## Tabular data 
::: {.callout-tip}
### At a glance... 

**Best for:** Discrete observations, station data, or summaries  
**Common formats:** CSV, TSV, Parquet, NetCDF tables
:::

For many users, tabular data feels familiar because it resembles spreadsheets used in tools like Excel or Google Sheets.

```

> | date       | latitude | longitude | value |

Reviewer 3: Maybe replace "value" with a geophysical variable name

A spreadsheet may show *what* the values are, but not *where* or *how* they were produced.

::: {.callout-note}
### Why JSON exists

JSON was developed as a lightweight structured text format for exchanging data between systems, particularly in web services and APIs. In Earth science, it is commonly used for metadata, configuration files, and small structured datasets rather than large scientific arrays.
:::

In general, tabular structures are well-suited for:

- Single-location time series
- Small collections of points
- Summaries and diagnostics

But are less suitable for:

- Continuous spatial fields
- Large-scale gridded products
- Analyses requiring detailed spatial context

## Vector data 

::: {.callout-tip}
### At a glance... 

**Best for:** Boundaries, tracks, and discrete geographic features  
**Common formats:** Shapefile, GeoJSON, GeoPackage, KML
:::

Vector data represent **discrete geographic features** rather than continuous spatial fields. 

```
• Point        ─── Line        ▢ Polygon
```

Examples:

- Glacier outlines
- Ice sheet boundaries
- Grounding lines

Unlike rasters:

- Geometry is explicit
- Detail depends on how the feature was digitized

> - Attributes are attached to features, not grid cells

Reviewer 3: Suggest delete.  I don't think the comparison between a vector feature and a grid cell is valid here.

### Geometry types

#### Points
```
•
```

Points represent a single location with no area or length.

Common uses:

- Observation stations
- Sample locations
- Reference points

#### Lines
```
───────
```

Lines represent linear features with length but no area.

Common uses:

- Grounding lines
- Ice front positions
- Transects

#### Polygons
```
▢▢▢▢
▢▢▢▢
```

Polygons represent areas with defined boundaries.

Common uses:

- Glacier outlines
- Ice sheet masks
- Drainage basins

> Polygons define *inside vs outside*, which is important for:

Reviewer: I'm not sure about adding "Polygons define inside vs outside" since that gets towards an analysis method and isn't a structure per se.

- Area calculations
- Masking raster data
- Regional summaries

### Attributes

Each feature includes attributed data:

```
Feature ID | Name        | Area_km2 | Year
-----------|-------------|----------|------
001        | Glacier A   | 12.4     | 2020
002        | Glacier B   |  8.9     | 2020
```
Importantly:

- Attributes describe properties of **features**

> - They do not imply values between features

Reviewer 3: Not sure I understand.

- Attributes are attached to features, not cells

Vector data represent **objects and boundaries**, not continuous fields.

::: {.callout-note}
### Why image formats exist in data portals

Formats like PNG, JPEG, and GIF are optimized for visualization rather than analysis. They are often used as previews and quicklook imagery so users can inspect spatial patterns without downloading large scientific datasets. 
:::

## Bonus: Resampling data

Resampling refers to transforming data between **different spatial or structural representations**, not simply changing file formats.

Common transformations include:

- swath → grid  
- grid → different grid  
- high resolution → low resolution  

Why resampling is unavoidable:

- Instruments observe along tracks, not grids
- Different products use different projections
- Analyses require common spatial frameworks

Every resampling step introduces assumptions:

- How values are aggregated
- How gaps are handled
- How boundaries are treated

These choices affect uncertainty and comparability and are documented in processing descriptions rather than inherent to any single format.

## Wrapping up

> 

Reviewer 4: I don't understand what this section is trying to convey.  I think of a data set lifecycle as the different versions of a data set, e.g. SPL3SMP v001 -> v009.

Do you mean the processing pipeline from telemetered raw data to the final data set?  but then including "tabular extraction for analysis" doesn't really fit.  Or is "lifecycle" meant to encompass:  [telemetered raw data] ---> [processing pipeline to prepare data for users] ---> [something a user might do to the data to get at something specific]?

> A single NSIDC dataset may involve multiple structures (and formats!) over its lifecycle:

Reviewer 3: Maybe describe in the context of a general RS image processing pipeline.  This could be tied to NASA processing levels.
